etna/packages/etna-js/hooks/useAsyncWork.js

```diff
@@ -25,7 +25,9 @@ export default function useAsyncWork(f, { cancelWhenChange = undefined }) {
   const [cancellable, setCancellable] = cancelWhenChange ? useState(new Cancellable()) : [null, null];
 
   if (cancellable) useEffect(function () {
-    cancellable.cancel();
+    // Return the cancel as a cleanup operation -- this will get invokved only between the cancelWhenChange has new
```

Comment:
Slight adjustment here.  useEffect can return a function that will be used _cleanup_ when the component dismounts,  We want to cancel any request chains after a component dismounts so that it does not attempt to change the state of a dismounted component.